### PR TITLE
Review migration

### PR DESCRIPTION
**Pull request by Tasneem52**
Added reviews controller, model(validations and associations), added reviews on show page of train controller, post reviews, added test for reviews.

### CODE REVIEW

File: app/javascript/react/components/Train.js [hunk 1]
```diff
@@ -25,13 +26,28 @@ class Train extends Component {
         this.setState( {
           train: response.train
         } )
+        console.log(this.state.train)
```
Comment (CassandraNewell) [inline app/javascript/react/components/Train.js]: Please ensure that all debugging statements are removed before pushing

File: spec/javascript/components/react_tests_spec.js [hunk 2]
```diff
@@ -11,7 +11,14 @@ describe('train component', () => {
 
   beforeEach(() => {
```
Comment (CassandraNewell) [inline spec/javascript/components/react_tests_spec.js:11]: Please rename this file -- your enzyme tests should be named by the component being tested

File: spec/javascript/components/react_tests_spec.js [hunk 3]
```diff
@@ -11,7 +11,14 @@ describe('train component', () => {
...
     jasmineEnzyme()
-    batTrain = {train: {id: 1, name: "Bat Train", description: "NaNaNaNaNa Bat Train", founding_year: "2020-03-27"} }
+    batTrain = {train:
+      {id: 1, name: "Bat Train", description: "NaNaNaNaNa Bat Train", founding_year: "2020-03-27",
```
Comment (CassandraNewell) [inline spec/javascript/components/react_tests_spec.js:14]: 😂

**Review by CassandraNewell — CHANGES_REQUESTED** [review]

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+class ReviewsController < ApplicationController
+  before_action :authorize_user
+
+
+  def new
+    @train = Train.find(params[:train_id])
+    @review = Review.new
+    @review.user = current_user
+
```
Comment (prophetsfall) [inline app/controllers/reviews_controller.rb]: please remove extra line

File: app/controllers/reviews_controller.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  def review_params
+    params.require(:review).permit(:rating, :description)
+  end
+
+  protected
+
+  def authorize_user
+    if !user_signed_in?
+      flash[:notice] = "You need to login or sign up to submit a review"
+        redirect_to '/users/sign_up'
```
Comment (prophetsfall) [inline app/controllers/reviews_controller.rb]: this line is indented one too far.

File: app/controllers/reviews_controller.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    end
+  end
+
+
```
Comment (prophetsfall) [inline app/controllers/reviews_controller.rb]: please remove both extra lines.

File: app/controllers/trains_controller.rb [hunk 7]
```diff
@@ -18,7 +19,7 @@ def create
     @train = Train.new(train_params)
```
Comment (prophetsfall) [inline app/controllers/trains_controller.rb]: please remove extra line

**Review by prophetsfall — CHANGES_REQUESTED** [review]
There are many extra lines through the first four files. Please review all files that were changed and ensure that indentation is correct and extra lines have been removed. Then resubmit and I will finish reviewing the actual code.

File: app/views/reviews/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<h1>Add a New Review for <%=@train.name%></h1>
+
+<%= form_for [@train, @review] do |f| %>
+  <%= f.label :rating, "Rating" %>
+  <%=  f.collection_radio_buttons(
+      :rating, [[1, '1'] ,[2, '2'],[3, '3'],[4, '4'],[5, '5']], :first, :last
+    ) do |b|
+      b.label { b.radio_button + b.text }
+      end %>
```
Comment (prophetsfall) [inline app/views/reviews/new.html.erb:9]: even though right about we use |f| as the variable for form using |b| for button here is a bit off convention. I'll leave it up to you whether to change it or not, but it's the kind of habit I would strongly advise against getting into.

**Review by prophetsfall — APPROVED** [review]
Please see my one comment and consider changing it on your next branch. This one is good to go.

**Pull request merged**